src/identity/components/GlobalHeader/IdentityUserAvatar.tsx

```diff
@@ -98,18 +100,22 @@ class IdentityUserAvatar extends React.Component<Props, State> {
         <div>
           {/* Button shape is ButtonShape.Square to make the height and width the same
             so we can use the border radius to make it a circle  */}
-          <Button
-            text={this.getInitials()}
-            shape={ButtonShape.Square}
-            color={
-              isPopoverOpen ? ComponentColor.Default : ComponentColor.Tertiary
-            }
-            onClick={this.togglePopoverState}
-            className={userAvatarButtonClassName}
-          />
+          {this.getInitials() && (
```

Comment:
Adding `{this.getInitials() &&` as a guard here resolves an error that can be thrown by clockface's `Button` component, which requires that either its `text` and `icon` props must be truthy.

While we have passed `<Button>` a `text` prop in this component, its value - `this.getInitials()` - can evaluate to falsy if the component renders before the user's name is retrieved from the backend. In that case, the default redux state is used (`firstName: ''`, `lastName: ''`), making `this.getInitials()` falsy, because `firstname.charAt(0)` + `lastName.chartAt(0)` will also evaluate to an empty string.

Reply:
Great catch!!